src/interfaces.rs

```diff
@@ -337,6 +347,23 @@ pub fn host_dispatch(
             host.process_events(unsafe { &*(ptr as *const api::Events) });
         }
 
+        OpCode::GetTime => {
+            return match host.get_time_info(value as i32) {
+                None => 0,
+                Some(result) => {
+                    thread_local! {
+                        static TIME_INFO: UnsafeCell<TimeInfo> =
+                            UnsafeCell::new(unsafe { mem::uninitialized() });
+                    }
+                    TIME_INFO.with(|time_info| {
+                        unsafe { *time_info.get() = result };
+                        time_info.get() as isize
```

Comment:
Why not `derive(Default)` on `TimeInfo` and use `Cell::new(TimeInfo::default())` and return `TIME_INFO.as_ptr()` ?
Then it would not require `unsafe`.